### PR TITLE
Install atom-package dependencies automatically

### LINKED ISSUE #116: Automatically install dependencies if not installed
To make it nicer for our users, install `autocomplete-plus` or `linter` if they aren't installed. 

See atom-typescript for ideas https://github.com/TypeStrong/atom-typescript/blob/eec59e111de711653d686794a872730d6994da0d/lib/main/atomts.ts#L237

### PR DESCRIPTION
**Pull request by sgwill**
Inspired by atom-typescript.

Closes #116

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,59 +1,59 @@
-{
```
Comment (nosami) [inline tsconfig.json]: This is just whitespace change... Looks like we need a .gitattributes file

**Comment by willl** [comment]
Open discussion... Q: If a user decides to remove OmniSharp Atom, should we or should we not remove dependencies?

File: tsconfig.json (hunk 1 above)
Comment (sgwill) [inline tsconfig.json]: Or I need to adjust my own settings

**Comment by nosami** [comment]
Not sure it would be easy to remove dependencies. You'd need to scan all the other plugins dependencies to make sure they were no longer needed.

**Comment by david-driscoll** [comment]
The question is, is there a "standard" in how packages state that they depend on one another?  I can't find any docs on anything like that.

If there is no standard, then we would have no easy way to tell if the dependency is in use or not anymore.

**Comment by sgwill** [comment]
At the moment there does not appear to be a standard.

**Comment by nosami** [comment]
Works well!

screen shot 2015-04-24 at 02 24 15
screen shot 2015-04-24 at 02 24 24

**Comment by ctolkien** [comment]
Looks awesome @sgwill  :star2:

**Pull request merged**